Rollbar/AspNetCore/RollbarMiddleware.cs

```diff
@@ -138,14 +165,34 @@ RequestDelegate nextRequestProcessor
                         RollbarLocator.RollbarInstance.Log(data);
                     }
 
-                    throw new Exception("The included internal exception processed by the Rollbar middleware", ex);
+                    throw new System.Exception("The included internal exception processed by the Rollbar middleware", ex);
                 }
                 finally
                 {
                     if (context != null && context.Response != null)
                     {
                         RollbarScope.Current.HttpContext.HttpAttributes.StatusCode = context.Response.StatusCode;
                     }
+
+                    if (TelemetryCollector.Instance.IsAutocollecting && context != null && context.Request != null)
+                    {
+                        telemetryMethod = context.Request.Method;
+                        telemetryUrl = context.Request.Host.Value + context.Request.Path;
+                        if (context.Response != null)
+                        {
+                            telemetryStatusCode = context.Response.StatusCode;
+                        }
+
+                        NetworkTelemetry networkTelemetry = new NetworkTelemetry(
+                            method: telemetryMethod,
+                            url: telemetryUrl,
+                            eventStart: telemetryStart,
+                            eventEnd: DateTime.Now,
+                            statusCode: telemetryStatusCode
+                            );
+                        TelemetryCollector.Instance.Capture(new Telemetry(TelemetrySource.Server, TelemetryLevel.Info, networkTelemetry));
```

Comment:
It appears that we are recording two events for each network request, is that right? We don't want to do that. We want only one with both the start and end time if possible, depending on potential race conditions with the network request and sending an item with that telemetry data. In JS we hold on to a reference to the underlying telemetry event and update the end time when the request is done. Most of the time that works, some times there is no end time in an event that gets sent for a variety of reasons.

The downside of doing it this way is that the UI will show two network events for the same request. I can think of some ways to mitigate this on the API side, but the way it is working now with JS is to assume that each network event is a distinct request.

Reply:
Yes it is still possible for this to be racey and two events could show up in the UI. We probably should think about this in the future and create some kind of correlation ID like you mention. I would prefer to just create telemetry items for each stage of the request and then link them together on the backend with some kind of ID rather than using racey mutable state, but let's save that for the future.

You are free to merge once you update, everything else looks good to me.

Reply:
@rokob, got it. Thanks! i'll update logic to one telemetry event per http request and merge it.